thirdparty/idstore/idstore.go

```diff
@@ -2,59 +2,78 @@ package idstore
 
 import (
 	"context"
+	"fmt"
 
 	mh "gx/ipfs/QmZyZDi491cCNTLfAhwcaDii2Kg4pwKRkhqQzURGDvY6ua/go-multihash"
 	bls "gx/ipfs/QmaG4DZ4JaqEfvPWt5nPPgoTzhc1tr1T3f4Nu9Jpdm8ymY/go-ipfs-blockstore"
 	cid "gx/ipfs/QmcZfnkapfECQGcLZaf9B79NRg7cRa9EnZh4LSbkCzwNvY/go-cid"
 	blocks "gx/ipfs/Qmej7nf81hi2x2tvjRBF3mcp74sQyuDH4VMYDGd1YtXjb2/go-block-format"
 )
 
+var MaxIdHashLen = 64
+
 // idstore wraps a BlockStore to add support for identity hashes
 type idstore struct {
 	bs bls.Blockstore
 }
 
-func IdStore(bs bls.Blockstore) bls.Blockstore {
+func IdStore(bs bls.Blockstore) *idstore {
```

Comment:
Any reason for exporting the implementation structure instead of the interface? It is generally better to export the interface as then it is clear what contract it follows.

Reply:
No.  Changed back.